portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -82,36 +84,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery commentHistory = datastore.prepare(commentHistoryQuery);
     List<Entity> commentHistoryList = commentHistory.asList(FetchOptions.Builder.withDefaults());
     int totalComments = commentHistoryList.size();
-    int totalPages;
 
     // Obtain the maximum number of comments to display. If the number is invalid, take the default value.
+    String maxCommentsToDisplayStr = request.getParameter("maxCommentsToDisplay");
     int maxCommentsToDisplay;
     try {
-      maxCommentsToDisplay = Integer.parseInt(request.getParameter("maxCommentsToDisplay"));
+      maxCommentsToDisplay = Integer.parseInt(maxCommentsToDisplayStr);
       if (maxCommentsToDisplay < 0) {
         maxCommentsToDisplay = DEFAULT_MAX_COMMENTS;
+        invalidMaxComments = true;
       }
     } catch (NumberFormatException e) {
       maxCommentsToDisplay = DEFAULT_MAX_COMMENTS;
+      // Upon initialization, the field is empty; this special case doesn't indicate erroneous user input.
+      if (!maxCommentsToDisplayStr.equals("")) {
```

Comment:
Got it! I'll switch to isEmpty(). It can only be a empty string.

Reply:
https://github.com/KathyFeiyang/step/pull/8#discussion_r435939281: I couldn't directly respond to that question so I'll post it here:

The overriding is intentional, because there might be situations where maxCommentsToDisplay=0 but the page ID user input is invalid. The code outside of the maxCommentsToDisplay=0 case checks for invalid page ID. But I agree that the logic is hard to read. It was a bit hard to write as well. I will recheck the code to fix any issues and also clean it up.